### PR TITLE
[WFLY-12418] Move to jakarta.security.enterprise:jakarta.security.ent…

### PR DESCRIPTION
**Pull request by skyllarr**
…erprise-api and org.glassfish.soteria:jakarta.security.enterprise

[WFLY-12418](https://issues.jboss.org/browse/WFLY-12418)

For now EE security will not be forked and there are jakarta releases already available that we can use.

### CODE REVIEW

File: feature-pack/src/license/full-feature-pack-licenses.xml [hunk 1]
```diff
@@ -685,8 +685,8 @@
       </licenses>
     </dependency>
     <dependency>
-      <groupId>javax.security.enterprise</groupId>
-      <artifactId>javax.security.enterprise-api</artifactId>
+      <groupId>jakarta.security.enterprise</groupId>
+      <artifactId>jakarta.security.enterprise-api</artifactId>
```
Comment (bstansberry) [inline feature-pack/src/license/full-feature-pack-licenses.xml]: The dependency entries in this file are kept in alphabetical order based on groupId then artifactId. So please move this one to the correct new spot.

File: ee-security/src/main/java/org/wildfly/extension/eesecurity/EESecurityDependencyProcessor.java [hunk 2]
```diff
@@ -40,7 +40,7 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
         final ModuleLoader moduleLoader = Module.getBootModuleLoader();
 
         final ModuleSpecification moduleSpec = unit.getAttachment(Attachments.MODULE_SPECIFICATION);
-        moduleSpec.addSystemDependency(new ModuleDependency(moduleLoader, ModuleIdentifier.fromString("javax.security.enterprise.api"), false, false, true, false));
+        moduleSpec.addSystemDependency(new ModuleDependency(moduleLoader, ModuleIdentifier.fromString("jakarta.security.enterprise.api"), false, false, true, false));
```
Comment (bstansberry) [inline ee-security/src/main/java/org/wildfly/extension/eesecurity/EESecurityDependencyProcessor.java]: Please leave the module name unchanged. This module provides classes in the javax.security.enterprise package. It isn't necessary to change the module name.

File: feature-pack/src/main/resources/modules/system/layers/base/jakarta/security/enterprise/api/main/module.xml [hunk 3]
```diff
@@ -22,13 +22,13 @@
   ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
   -->
 
-<module xmlns="urn:jboss:module:1.5" name="javax.security.enterprise.api">
+<module xmlns="urn:jboss:module:1.5" name="jakarta.security.enterprise.api">
```
Comment (bstansberry) [inline feature-pack/src/main/resources/modules/system/layers/base/jakarta/security/enterprise/api/main/module.xml]: Please leave the module name (and therefore the file name) unchanged.

File: feature-pack/src/main/resources/modules/system/layers/base/org/glassfish/soteria/main/module.xml [hunk 4]
```diff
@@ -28,11 +28,11 @@
     </properties>
 
     <resources>
-        <artifact name="${org.glassfish.soteria:javax.security.enterprise}"/>
+        <artifact name="${org.glassfish.soteria:jakarta.security.enterprise}"/>
     </resources>
 
     <dependencies>
-        <module name="javax.security.enterprise.api" />
+        <module name="jakarta.security.enterprise.api" />
```
Comment (bstansberry) [inline feature-pack/src/main/resources/modules/system/layers/base/org/glassfish/soteria/main/module.xml]: Please leave the module name unchanged.

File: feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/ee-security/main/module.xml [hunk 5]
```diff
@@ -46,7 +46,7 @@
         <module name="org.jboss.msc"/>
         <module name="org.jboss.logging"/>
         <module name="org.glassfish.soteria" />
-        <module name="javax.security.enterprise.api" />
+        <module name="jakarta.security.enterprise.api" />
```
Comment (bstansberry) [inline feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/ee-security/main/module.xml]: Please leave the module name unchanged.

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: ee-security/src/main/java/org/wildfly/extension/eesecurity/EESecurityDependencyProcessor.java (hunk 2 above)
Comment (Skyllarr) [reply ee-security/src/main/java/org/wildfly/extension/eesecurity/EESecurityDependencyProcessor.java]: Thank you. Fixed.

File: feature-pack/src/license/full-feature-pack-licenses.xml (hunk 1 above)
Comment (Skyllarr) [reply feature-pack/src/license/full-feature-pack-licenses.xml]: Fixed.

File: feature-pack/src/main/resources/modules/system/layers/base/jakarta/security/enterprise/api/main/module.xml (hunk 3 above)
Comment (Skyllarr) [reply feature-pack/src/main/resources/modules/system/layers/base/jakarta/security/enterprise/api/main/module.xml]: Fixed.

File: feature-pack/src/main/resources/modules/system/layers/base/org/glassfish/soteria/main/module.xml (hunk 4 above)
Comment (Skyllarr) [reply feature-pack/src/main/resources/modules/system/layers/base/org/glassfish/soteria/main/module.xml]: Fixed.

File: feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/ee-security/main/module.xml (hunk 5 above)
Comment (Skyllarr) [reply feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/extension/ee-security/main/module.xml]: Fixed.

**Review by bstansberry — APPROVED** [review]
:-)

**Pull request merged**